### PR TITLE
use arg parser to parse input options

### PR DESCRIPTION
**Pull request by jonahwilliams**
Since we'll need to add more args in the future for things like relative heights and tessellation options

### CODE REVIEW

File: packages/vector_graphics_compiler/pubspec.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ dependencies:
   meta: ^1.7.0
   path_parsing: ^1.0.0
   xml: ^5.3.1
+  args: ^2.3.0
```
Comment (dnfield) [inline packages/vector_graphics_compiler/pubspec.yaml:17]: Would it make sense to have this be a dev dependency?  Does that even work?

Ideally it'd only be a dependency for the files in bin.

Comment (jonahwilliams) [reply packages/vector_graphics_compiler/pubspec.yaml]: Oh, you're right this can be a dev_dependency

Comment (jonahwilliams) [reply packages/vector_graphics_compiler/pubspec.yaml]: fixed!

File: packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+    print(err.message);
+    print(argParser.usage);
     exit(1);
   }
-  final String xml = File(args[0]).readAsStringSync();
-  final File outputFile = File(args[1]);
+  final String inputFilePath = results['input'] as String;
+  final String xml = File(inputFilePath).readAsStringSync();
+  final File outputFile =
+      File(results['output'] as String? ?? '$inputFilePath.vg');
```
Comment (dnfield) [inline packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:32]:
```suggestion
      File(results['output'] as String? ?? '${inputFilePath.replaceAll('.svg', '.vg'}');
```

**Review by dnfield — APPROVED** [review]

File: packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:32]: what if the input file is named `foo.svgProcessing.svg`, then we'd write `foo.vgProcessing.vg` its not wrong exactly but appending is safer if the user doesn't care what the output is exactly.

Comment (dnfield) [reply packages/vector_graphics_compiler/bin/vector_graphics_compiler.dart:32]: Yeah, fair. You could try to make it just replace the last .svg I guess but it's not worth the complexity.

**Pull request merged**